src/deprecated_library/castleengine.lpr

```diff
@@ -910,53 +911,431 @@ function CGE_GetVariableInt(eVar: cInt32): cInt32; cdecl;
   end;
 end;
 
-procedure CGE_SetNodeFieldValue(szNodeName, szFieldName: pcchar;
-                                fVal1, fVal2, fVal3, fVal4: cFloat); cdecl;
+procedure CGE_SetNodeFieldValue_SFFloat(szNodeName, szFieldName: pcchar; value: cFloat); cdecl;
 var
   aField: TX3DField;
 begin
   try
-    if not CGE_VerifyScene('CGE_SetNodeFieldValue') then exit;
+    if not CGE_VerifyScene('CGE_SetNodeFieldValue_SFFloat') then exit;
+    aField := MainScene.Field(PChar(szNodeName), PChar(szFieldName));
+    if (aField <> nil) and (aField is TSFFloat) then
+       TSFFloat(aField).Send(value);
```

Comment:
We need to have a way to notify developer in case the type is incorrect. I.e. right now the `CGE_SetNodeFieldValue_SFFloat` will just silently not do anything (no warning, no error), if it will be used for X3D type other than `SFFloat`.

My proposal: just use "as" cast, and let it raise an exception (which you catch around this block already, so it will result in a nice message) when the type is incorrect.

So instead of 

```delphi
    if (aField <> nil) and (aField is TSFFloat) then
       TSFFloat(aField).Send(value);
```

->

```delphi
    if aField <> nil then
       (aField as TSFFloat).Send(value);
```